ambry-store/src/main/java/com.github.ambry.store/Log.java

```diff
@@ -73,6 +77,7 @@
       throw new IOException("Could not read from directory: " + dataDir);
     } else {
       initialize(getSegmentsToLoad(segmentFiles), segmentCapacityInBytes);
+      this.isLogSegmented = isExistingLogSegmented();
```

Comment:
may be in future. since we have this "isLogSegmented" we might re-use this elsewhere in this Log whereever it makes sense